### ISSUE TITLE
CIDER functionality does not work for code in JARs anymore

### ISSUE THREAD

**Issue by hanshuebner** [clojure-emacs/cider#2481]
## Expected behavior

`C-c t v` on a function that lives in a jar file should toggle tracing, like it does for functions defined in local files.

## Actual behavior

Error message is shown:
```
‘cider-toggle-trace-var’ requires the nREPL op "toggle-trace-var".  Please, install (or update) cider-nrepl 0.18.0 and restart CIDER
```

## Steps to reproduce the problem

`M-.` to the source code of a function that is defined in a .jar file, then type `C-c t v` on the function name.  Alternatively, find a clojure function that is call from the source file loaded from the jar, and press `M-.` on it, this yields the message

```
‘cider-find-var’ requires the nREPL op "info".  Please, install (or update) cider-nrepl 0.18.0 and restart CIDER
```

### CIDER version information

```
;; CIDER 0.18.0 (Saigon), nREPL 0.2.13
;; Clojure 1.8.0, Java 1.8.0_51
```

### Lein/Boot version

`Leiningen 2.8.1 on Java 1.8.0_51 Java HotSpot(TM) 64-Bit Server VM`

### Emacs version

`GNU Emacs 25.2.1 (x86_64-unknown-linux-gnu) of 2017-06-30`

### Operating system

CentOS 7

**Comment by bbatsov** [comment]
Hmm, that's pretty weird. I'd expect for you to get some missing sesman-session error in such a case, but I can't imagine why the middleware'd be missing. That seems relevant to what we're trying to fix in #2483. //cc @vspinu

**Comment by vspinu** [comment]
Seems to be unrelated to session management. In the current master sessions are auto-linked to jars. That error might be a cover-up irrelevant to the core-issue. What does your `*nrepl-messages*` buffer say?

For me `cider-toggle-trace-var` either time-outs when called on a clojure.core var or throws `(wrong-type-argument stringp nil)`, so something is clearly amiss in there.

**Comment by hanshuebner** [comment]
@vspinu I don't have a `*nrepl-messages*` buffer, maybe because I'm using `M-x cider-jack-in`?  Let me know how I can assist if I can.

**Comment by vspinu** [comment]
`(setq nrepl-log-messages t)`

**Comment by hanshuebner** [comment]
When I type `M-.` in a source file to navigate to the implementation of
`map?`, this is written to the `*nrepl-messages ...` buffer:
```
(<--
  id           "19"
  session      "b641abbb-ba5d-4dd4-8d70-740ed8737e2b"
  time-stamp   "2018-10-12 20:59:34.526307859"
  added        "1.0"
  arglists-str "[x]"
  column       1
  doc          "Return true if x implements IPersistentMap"
  file         "jar:file:/home/hans/.m2/repository/org/clojure/clojure/
1.8.0..."
  line         167
  name         "map?"
  ns           "clojure.core"
  resource     "clojure/core.clj"
  see-also     ("clojure.core/hash-map" "clojure.core/set?"
"clojure.core/vector?")
  static       "true"
  status       ("done")
)
```
When I then use `M-.` inside of the code from the jar, nothing else shows
up in `*nrepl-messages ...`.

**Comment by kommen** [comment]
Just throwing it in here, but this could also be an issue since #2467

**Comment by kommen** [comment]
Although `cider-toggle-trace-var` just works for me for `defn`s in files inside jars...

**Comment by jeremyheiler** [comment]
This issue:

```
user-error: ‘cider-find-var’ requires the nREPL op "info".  Please, install (or update) cider-nrepl 0.18.0 and restart CIDER
```

Also appears to exist for source files that are not within the directory that the REPL was started in.

That is, I have a `project.clj` that has additional source paths from a "cousin" directory.

**Comment by bbatsov** [comment]
I'm pretty sure that's a regression from #2467.

**Comment by bbatsov** [comment]
(someone should debug what's happening in the `cider-ensure-nrepl-op` function)

**Comment by jeremyheiler** [comment]
I did some preliminary debugging, and tracing through these functions:

```
cider-nrepl-op-supported-p
cider-current-repl
cider-repls
sessman-current-session
```

The call to `(sesman-current-session 'CIDER)` returns `nil`.

**Comment by vspinu** [comment]
Can you step through `sesman-friendly-session-p` and see what's going on? Looks like your jar is not in the classpath. Should be trivial to track down. Path mismatch of some kind.

**Comment by kommen** [comment]
For what it is worth, I just tried to reproduce the issue, but toggling tracing with `cider-toggle-trace-var` works for me.

```
;; CIDER 0.19.0snapshot (package: 20181019.812), nREPL 0.4.5
;; Clojure 1.9.0, Java 1.8.0_102
```

Navigated to a clj source file of a maven dependency of a project using deps.edn, and in the clj file inside the jar `cider-toggle-trace-var` worked as expected.

**Comment by jeremyheiler** [comment]
I believe this commit has fixed the issue for me: e9108a03266e58eef28b04a8b2ad5209355ee953.

Thank you @vspinu

**Comment by vspinu** [comment]
Right. This one should be fixed. @hanshuebner, @kommen could you please confirm?

**Comment by kommen** [comment]
Didn't test if that commit in particular fixed it, but it at least works for me.™️  :)
